internal/controller/etcdopstask/handler/ondemandsnapshot/ondemandsnapshot.go

```diff
@@ -0,0 +1,224 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ondemandsnapshot
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/common"
+	"github.com/gardener/etcd-druid/internal/controller/etcdopstask/handler"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ErrGetEtcd represents the error in case of fetching etcd object. Reason could be internal error
+	ErrGetEtcd druidv1alpha1.ErrorCode = "ERR_GET_ETCD"
+	// ErrEtcdNotReady represents the error in case etcd is not ready
+	ErrEtcdNotReady druidv1alpha1.ErrorCode = "ERR_ETCD_NOT_READY"
+	// ErrBackupNotEnabled represents the error in case backup is not enabled
+	ErrBackupNotEnabled druidv1alpha1.ErrorCode = "ERR_BACKUP_NOT_ENABLED"
+	// ErrGetCASecret represents the error in case of failure in fetching the secret
+	ErrGetCASecret druidv1alpha1.ErrorCode = "ERR_GET_SECRET"
+	// ErrCreateHTTPRequest represents the error in case of failure in creating http request
+	ErrCreateHTTPRequest druidv1alpha1.ErrorCode = "ERR_CREATE_HTTP_REQUEST"
+	// ErrExecuteHTTPRequest represents the error in case of failure in executing http request
+	ErrExecuteHTTPRequest druidv1alpha1.ErrorCode = "ERR_EXECUTE_HTTP_REQUEST"
+	// ErrCreateSnapshot represents the error in case of failure in creating snapshot
+	ErrCreateSnapshot druidv1alpha1.ErrorCode = "ERR_CREATE_SNAPSHOT"
+)
+
+// Handler implements the task.Handler interface for handling on-demand snapshot tasks.
+type Handler struct {
+	client        client.Client
```

Comment:
rename `client` to `k8sClient` as there is also a `httpClient` field in the same struct.